nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -89,17 +89,19 @@ def __parse_item(line: str, manifest_file: str) -> Dict[str, Any]:
             f"Manifest file {manifest_file} has invalid json line structure: {line} without proper audio file key."
         )
 
-    # If the audio path is relative, and not using tarred dataset,
-    # attach the parent directory of manifest to the audio path.
-    # Assume "audio_file" starts with a dir, such as "wavs/xxxxx.wav".
-    # If using a tarred dataset, the "audio_path" is like "_home_data_tarred_wavs_xxxx.wav",
-    # so we will just ignore it.
+    # If the audio path is a relative path and does not exist,
+    # try to attach the parent directory of manifest to the audio path.
+    # Revert to the original path if the new path still doesn't exist.
+    # Assume that the audio path is like "wavs/xxxxxx.wav".
     manifest_dir = Path(manifest_file).parent
     audio_file = Path(item['audio_file'])
-    if not audio_file.is_file() and not audio_file.is_absolute() and audio_file.parent != Path("."):
-        # assume the wavs/ dir and manifest are under the same parent dir
+    if not audio_file.is_file() and not audio_file.is_absolute():
```

Comment:
To note, I think even this is_file() check will fail for too long file names. Any idea ?

Reply:
I think we can try to avoid too long file names. When creating tarred datasets, we can trim the common path of all audio files and use the trimmed paths as the new file names.  

For example,
```
/a/b/c/d/e/xxxxxx.wav
/a/b/c/d/f/yyyyyy.wav
```
can be trimmed as 
```
e/xxxxxx.wav
f/yyyyyy.wav
```

Reply:
No let's not do that, there are many datasets of ASR in mcc and MLS where there is a lot of common folders.

Reply:
How about changing the code to the following? We just keep the original path if the input file is 255 chars or longer.
```python
 if (len(str(audio_file) < 255) and not audio_file.is_file() and not audio_file.is_absolute():
```

Reply:
There might be some confusion here (or maybe I am the confused one!). The 255 character limit refers to individual "nodes" in the file system tree. You can certainly have a path to a file with a  length greater than 255, assuming there are directories leading to it. It's just that each file name and directory name must individually not exceed 255 characters.

Reply:
Yes, I understand.